src/Form/FormField.js

```diff
@@ -154,39 +173,94 @@ export default class FormField extends Component<Props> {
       ...otherRootProps
     };
 
+    // Label structure differs if input is a group - e.g. RadioGroup, CheckboxGroup
+    const groupRole = this.getGroupRole();
+    const isGroup = Boolean(groupRole);
+    const Label = isGroup ? 'div' : 'label';
+
+    const textWrapperProps = {
+      hideLabel,
+      key: `${this.id}-textWrapper`
+    };
+
+    const labelTextProps = {
+      id: `${this.id}-labelText`
+    };
+
     const captionProps = {
       caption,
       variant,
       id: `${this.id}-caption`
     };
 
-    const controlProps = {
-      'aria-describedby': caption && captionProps.id,
-      required,
-      variant,
-      // Note: Props are spread to input rather than Root
-      ...restProps
+    const controlProps = (props = {}) => {
+      return {
+        'aria-describedby': caption && captionProps.id,
+        key: `${this.id}-control`,
+        required,
+        rootProps: isGroup
+          ? {
+              'aria-labelledby': labelTextProps.id,
+              'aria-describedby': caption && captionProps.id,
+              role: groupRole,
+              ...props.rootProps
+            }
+          : props.rootProps,
+        variant,
+        // Note: Props are spread to input rather than Root
+        ...restProps
+      };
     };
 
-    const control = input
-      ? createElement(input, controlProps)
-      : children ? cloneElement(Children.only(children), controlProps) : null;
+    let control = null;
+    if (input) {
+      control = createElement(input, controlProps());
+    } else if (children) {
+      const child = Children.only(children);
+      control = cloneElement(child, controlProps(child.props));
+    }
 
     return (
       <Root {...rootProps}>
-        <label>
-          <TextWrapper hideLabel={hideLabel}>
-            {label}
+        <Label>
+          <TextWrapper {...textWrapperProps}>
+            <span {...labelTextProps}>{label}</span>
             {(required || secondaryText) && (
               <SecondaryText secondaryText={secondaryText}>
                 {secondaryText ? secondaryText : requiredText}
               </SecondaryText>
             )}
           </TextWrapper>
           {control}
-        </label>
+        </Label>
         {caption && <Caption {...captionProps}>{caption}</Caption>}
       </Root>
     );
   }
+
+  getGroupRole = () => {
+    const { children, input, isGroup, isRadioGroup } = this.props;
+    let groupRole;
+    if (isRadioGroup) {
+      groupRole = 'radiogroup';
+    } else if (isGroup) {
+      groupRole = 'group';
+    } else {
+      if (input && input.name) {
+        groupRole = this.getGroupRoleFromName(input.name);
+      } else if (children) {
+        const child = Children.only(children);
+        if (child.type && child.type.name) {
+          groupRole = this.getGroupRoleFromName(child.type.name);
+        }
+      }
+    }
+    return groupRole;
+  };
+
+  getGroupRoleFromName(name: string) {
+    return REGEX_RADIO_GROUP.test(name)
```

Comment:
this complexity makes me think we should take a harder look at having separate components for adding labels and roles to our Form elements, ie: <RadioField>, <RadioGroupField>, <TextField> etc.

Reply:
The code has been simplified.  Really it is not that complex though.  FormField just needs to see if the associated input is a grouped control like a radio or a checkbox, and if so, adjust the label structure.  This is because things like radios and checkboxes have labels of their own and labels can't be nested.  They need to use fieldset/legend or the equivalent role="group/radiogroup" and aria-labelledby.